diff --git a/regexp.c b/regexp.c
index 309673f..15e8f1c 100644
--- a/regexp.c
+++ b/regexp.c
@@ -17,11 +17,13 @@
 #include "builtins.h"
 
 /* all the options of interest for regex functions */
+#if PG_VERSION_NUM < 180000
 typedef struct pg_re_flags
 {
 	int			cflags;			/* compile flags for Spencer's regex code */
 	bool		glob;			/* do it globally (for each occurrence) */
 } pg_re_flags;
+#endif
 
 /* cross-call state for regexp_match and regexp_split functions */
 typedef struct regexp_matches_ctx
